app/questionnaire/navigation.py

```diff
@@ -31,110 +31,143 @@ def build_navigation(self, current_group_id, current_group_instance):
+                    and self._can_reach_summary_confirmation(last_group_id, last_block_id) is False:
+                continue
 
-            skip_group = self._should_skip_group(current_group_instance, group)
-            if not skip_group:
-                repeating_rule = SchemaHelper.get_repeat_rule(group)
-                if repeating_rule:
-                    logger.debug("building repeating navigation", group_id=group['id'])
-                    navigation.extend(self._build_repeating_navigation(repeating_rule, group, current_group_id,
-                                                                       current_group_instance))
-                elif last_group_id == group['id'] and can_get_to_summary:
-                    logger.debug("building navigation", group_id=group['id'])
-                    if len(self.completed_blocks) > 0 and set(self.routing_path[:-1]).issubset(self.completed_blocks):
-                        navigation.append(self._build_single_navigation(group, current_group_id, first_location))
-                elif last_group_id != group['id'] and not is_summary_or_confirm_group:
-                    logger.debug("building navigation", group_id=group['id'])
-                    navigation.append(self._build_single_navigation(group, current_group_id, first_location))
+            repeating_rule = SchemaHelper.get_repeat_rule(
+                SchemaHelper.get_group(self.survey_json, first_group_id))
+            if repeating_rule:
+                navigation.extend(self._build_repeating_navigation(repeating_rule, section, current_group_id,
+                                                                   current_group_instance, last_block_id))
+            else:
+                navigation.append(self._build_single_navigation(section, current_group_id,
+                                                                Location(first_group_id, 0, first_block_id),
+                                                                last_block_id))
 
         return navigation
 
-    def _should_skip_group(self, current_group_instance, group):
+    def _get_non_skipped_groups(self, section):
+        non_skipped_groups = []
+        for group_id in section['group_order']:
+            if not self._should_skip_group(group_id):
+                non_skipped_groups.append(group_id)
+
+        return non_skipped_groups
 
-        skip_group = False
+    def _should_skip_group(self, group_id):
+        skip_nav = False
+        group = SchemaHelper.get_group(self.survey_json, group_id)
         skip_conditions = SchemaHelper.get_skip_condition(group)
 
         if skip_conditions:
-            skip_group = evaluate_skip_conditions(skip_conditions, self.metadata, self.answer_store,
-                                                  current_group_instance)
-        return skip_group
+            skip_nav = evaluate_skip_conditions(skip_conditions, self.metadata, self.answer_store)
+
+        return skip_nav
```

Comment:
`skip_nav` should be renamed something like `skip_group` as this method doesn't know anything about navigation